sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -340,7 +340,7 @@ static int32_t _az_http_client_curl_upload_read_callback(
   // Update the userdata span
   //  ptr will point to remaining data to be copied
   //  length and capacity are set to the size of the remaining content
-  *upload_content = az_span_slice(*upload_content, dst_buffer_size, -1);
+  *upload_content = az_span_slice(*upload_content, dst_buffer_size, az_span_length(*upload_content));
```

Comment:
This makes me nervous. -1 should be usable here. Did your changes break the meaning of -1?

Reply:
I will undo this change. It is not that I wanted to change the meaning of `-1`, I just wanted the spans that were created from string literals to have one extra byte of capacity, for the 0 character that is always there for the string literals.

Reply:
Undone.